src/models.zig

```diff
@@ -149,6 +161,296 @@ pub const Model = struct {
+					try vertWriter.print("v {d:.4} {d:.4} {d:.4}\n", .{q.corners[2][0] - 0.5 + q.normal[0], q.corners[2][2] + q.normal[2], -(q.corners[2][1] - 0.5 + q.normal[1])});
+					try vertWriter.print("v {d:.4} {d:.4} {d:.4}\n", .{q.corners[3][0] - 0.5 + q.normal[0], q.corners[3][2] + q.normal[2], -(q.corners[3][1] - 0.5 + q.normal[1])});
+
+					try uvWriter.print("vt {d:.4} {d:.4}\n", .{(q.cornerUV[0][0] + textureSlotX) / 4.0, (q.cornerUV[0][1] + textureSlotY) / 4.0});
+					try uvWriter.print("vt {d:.4} {d:.4}\n", .{(q.cornerUV[1][0] + textureSlotX) / 4.0, (q.cornerUV[1][1] + textureSlotY) / 4.0});
+					try uvWriter.print("vt {d:.4} {d:.4}\n", .{(q.cornerUV[2][0] + textureSlotX) / 4.0, (q.cornerUV[2][1] + textureSlotY) / 4.0});
+					try uvWriter.print("vt {d:.4} {d:.4}\n", .{(q.cornerUV[3][0] + textureSlotX) / 4.0, (q.cornerUV[3][1] + textureSlotY) / 4.0});
+
+					try normWriter.print("vn {d:.4} {d:.4} {d:.4}\n", .{q.normal[0], q.normal[2], -q.normal[1]});
+
+					try faceWriter.print("f {}/{}/{} {}/{}/{} {}/{}/{} {}/{}/{}\n", .{vertInd, vertInd, normInd, vertInd + 1, vertInd + 1, normInd, vertInd + 2, vertInd + 2, normInd, vertInd + 3, vertInd + 3, normInd});
+					
+					vertInd += 4;
+					normInd += 1;
+				}
+			}
+		}
+
+		var data = std.ArrayList(u8).init(main.stackAllocator.allocator);
+		defer data.deinit();
+		var dataWriter = data.writer();
+
+		try dataWriter.writeAll(vertData.items);
+		try dataWriter.writeAll(uvData.items);
+		try dataWriter.writeAll(normData.items);
+		try dataWriter.writeAll(faceData.items);
+
+		try main.files.write(path, data.items);
+	}
+
+	pub fn loadModel(data: []const u8) !u16 {
+		var vertices = std.ArrayList(Vec3f).init(main.stackAllocator.allocator);
+		defer vertices.deinit();
+
+		var normals = std.ArrayList(Vec3f).init(main.stackAllocator.allocator);
+		defer normals.deinit();
+
+		var uvs = std.ArrayList(Vec2f).init(main.stackAllocator.allocator);
+		defer uvs.deinit();
+
+		var tris = std.ArrayList(Triangle).init(main.stackAllocator.allocator);
+		defer tris.deinit();
+
+		var quadFaces = std.ArrayList(Quad).init(main.stackAllocator.allocator);
+		defer quadFaces.deinit();
+
+		var fixed_buffer = std.io.fixedBufferStream(data);
+		var buf_reader = std.io.bufferedReader(fixed_buffer.reader());
+		var in_stream = buf_reader.reader();
+		var buf: [128]u8 = undefined;
+		while (try in_stream.readUntilDelimiterOrEof(&buf, '\n')) |lineUntrimmed| {
+			if (lineUntrimmed.len < 3)
+				continue;
+			
+			var line = lineUntrimmed;
+			if (line[line.len - 1] == '\r') {
+				line = line[0..line.len - 1];
+			}
+			
+			if (std.mem.eql(u8, line[0..1], "#"))
+				continue;
+			
+			if (std.mem.eql(u8, line[0..2], "v ")) {
+				var coordsIter = std.mem.split(u8, line[2..], " ");
+				var coords: Vec3f = undefined;
+				var i: usize = 0;
+				while (coordsIter.next()) |coord| : (i += 1) {
+					coords[i] = try std.fmt.parseFloat(f32, coord);
```

Comment:
Instead of blindly returning the error, it should print line number, error name and the value it tried to parse.
Then it should continue with a default value, since we don't want world loading to fail on minor problems like this.